offload `fatal error: 'OffloadAPI.h' file not found`  - on linux

```
FAILED: offload/liboffload/CMakeFiles/LLVMOffload.dir/src/OffloadImpl.cpp.o
/build/source/build-llvm/./bin/clang++ --target=x86_64-pc-linux-gnu -DDEBUG_PREFIX=\"Liboffload\" -DTARGET_NAME=\"Liboffload\" -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/source/llvm/include -I/build/source/build-llvm/include -I/build/source/build-llvm/runtimes/runtimes-bins/offload/liboffload/API -I/build/source/build-llvm/runtimes/runtimes-bins/offload/liboffload/../include -I/build/source/offload/liboffload/include -I/build/source/offload/liboffload/../include -I/build/source/offload/liboffload/../plugins-nextgen/common/include -I/build/source/build-llvm/runtimes/runtimes-bins/offload/liboffload/../plugins-nextgen/common/include -I/build/source/offload/plugins-nextgen/common/include -I/build/source/build-llvm/runtimes/runtimes-bins/offload/plugins-nextgen/common/include -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=3 -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fdebug-prefix-map=/build/source/build-llvm/runtimes/runtimes-bins=../../../ -fdebug-prefix-map=/build/source/= -no-canonical-prefixes -ffile-prefix-map=/build/source/build-llvm/runtimes/runtimes-bins=../../../ -ffile-prefix-map=/build/source/= -no-canonical-prefixes -O3 -DNDEBUG -std=c++17 -fPIC  -fno-exceptions -funwind-tables -fno-exceptions -Werror=global-constructors -MD -MT offload/liboffload/CMakeFiles/LLVMOffload.dir/src/OffloadImpl.cpp.o -MF offload/liboffload/CMakeFiles/LLVMOffload.dir/src/OffloadImpl.cpp.o.d -o offload/liboffload/CMakeFiles/LLVMOffload.dir/src/OffloadImpl.cpp.o -c /build/source/offload/liboffload/src/OffloadImpl.cpp
In file included from /build/source/offload/liboffload/src/OffloadImpl.cpp:14:
/build/source/offload/liboffload/include/OffloadImpl.hpp:11:10: fatal error: 'OffloadAPI.h' file not found
   11 | #include <OffloadAPI.h>
      |          ^~~~~~~~~~~~~~
1 error generated.
```
probably caused by #141982

--- a/offload/liboffload/API/CMakeLists.txt
+++ b/offload/liboffload/API/CMakeLists.txt
@@ -42,5 +42,7 @@ else()
             COMMAND ${CMAKE_COMMAND} -E copy_if_different generated/${file}.gen ${CMAKE_CURRENT_BINARY_DIR}/${file}
             DEPENDS generated/${file}.gen
         )
+        add_custom_target(OffloadAPI.${file} DEPENDS ${file})
+        add_dependencies(OffloadAPI OffloadAPI.${file})
     endforeach()
 endif()

